### PR TITLE
refactor(tests/e2e): simplify `openPopup` helper

### PR DESCRIPTION
**Pull request by sidvishnoi**
Discovered when writing article for end-to-end testing 😄

### CODE REVIEW

**Review by sidvishnoi — COMMENTED** [review]
test-e2e

File: tests/e2e/pages/popup.ts [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 export { connectWallet } from '../helpers/testWallet';
 
 export async function openPopup(
   context: BrowserContext,
-  { browserName, channel }: BrowserInfo,
-  extensionId: string,
+  background: Background,
 ) {
-  const url = getPopupUrl({ browserName, channel }, extensionId);
+  const url = await background.evaluate(() => chrome.action.getPopup({}));
```
[the remark this answers is not in the document]
Comment (sidvishnoi) [reply tests/e2e/pages/popup.ts:12]: That'll require importing polyfill in each evaluate context.

[the remark this answers is not in the document]
Comment (sidvishnoi) [reply tests/e2e/pages/popup.ts:12]: Will check if Playwright lets us define such polyfills. But `chrome` is supposed to work in Edge/Chrome and Firefox.

[the remark this answers is not in the document]
Comment (sidvishnoi) [reply tests/e2e/pages/popup.ts:12]: I couldn't find a way for now, but something like `exposeBinding` would be the way. Or we can make `browser` a global in background.js

Something of a general problem, I think I will check that as a followup.

**Review by sidvishnoi — COMMENTED** [review]
test-e2e

**Review by sidvishnoi — COMMENTED** [review]
test-e2e

**Review by sidvishnoi — COMMENTED** [review]
test-e2e

**Review by sidvishnoi — COMMENTED** [review]
test-e2e

**Review by sidvishnoi — COMMENTED** [review]
test-e2e

**Comment by sidvishnoi** [comment]
The failures on MacOS happen even without this change. Debugging. Day 3.

**Review by raducristianpopa — COMMENTED** [review]
Lets push the fix for macOS in another PR since the issue was happening before as well.

**Review by raducristianpopa — APPROVED** [review]

**Review by sidvishnoi — COMMENTED** [review]
test-e2e

**Pull request merged**